src/main/java/org/mastodon/blender/csv/ColorSchemeDialog.java

```diff
@@ -0,0 +1,153 @@
+
+/**
+ * A JComboBox for selecting a color scheme.
+ * <br>
+ * This class provides can show an option dialog to let the user select a coloring scheme from a list,
+ * including tag sets and feature color modes.
+ */
+public class ColorSchemeDialog
+{
+	private ColorSchemeDialog()
+	{
+		// prevent instantiation
+	}
+
+	private static final String DEFAULT_OPTION = "[no coloring scheme]";
+
+	private static final String TAGS_HEADER = "Tags:";
+
+	private static final String FEATURE_COLOR_MODES_HEADER = "Feature Color Modes:";
+
+	private static final String SEPARATOR = "-----------------";
+
+	/**
+	 * Show a dialog to let the user select a coloring scheme.
+	 * @param projectModel the project model
+	 * @return the name of the selected coloring scheme, or {@code null} if none was selected.
+	 */
+	public static String showDialog( ProjectModel projectModel )
+	{
+		JComboBox< ColorFunction > comboBox = getColorSchemesComboBox( projectModel );
+
+		int result = JOptionPane.showOptionDialog(
+				null,
+				comboBox,
+				"Select coloring scheme",
+				JOptionPane.OK_CANCEL_OPTION,
+				JOptionPane.PLAIN_MESSAGE,
+				null,
+				null,
+				DEFAULT_OPTION
+		);
+
+		if ( result == JOptionPane.OK_OPTION )
+		{
+			ColorFunction selectedItem = ( ColorFunction ) comboBox.getSelectedItem();
+			if ( selectedItem == null )
+				return null;
+			if ( selectedItem.getGroup() == null )
+				return null;
+			return selectedItem.toString();
+		}
+		return null;
+	}
+
+	private static JComboBox< ColorFunction > getColorSchemesComboBox( ProjectModel projectModel )
+	{
+		List< ColorFunction > options = getColorSchemeOptions( projectModel );
+		JComboBox< ColorFunction > comboBox = new JComboBox<>( options.toArray( new ColorFunction[ 0 ] ) );
+		comboBox.setRenderer( new ColorSchemeOptionsRenderer() );
+		return comboBox;
+	}
+
+	static List< ColorFunction > getColorSchemeOptions( ProjectModel projectModel )
+	{
+		List< ColorFunction > colorFunctions = new ArrayList<>();
+		colorFunctions.add( new EmptyColorFunction( DEFAULT_OPTION ) );
+		colorFunctions.add( new EmptyColorFunction( SEPARATOR ) );
+		colorFunctions.add( new EmptyColorFunction( TAGS_HEADER ) );
+		colorFunctions.addAll( GraphToCsvUtils.getTagSetColorFunctions( projectModel ) );
+		colorFunctions.add( new EmptyColorFunction( SEPARATOR ) );
+		colorFunctions.add( new EmptyColorFunction( FEATURE_COLOR_MODES_HEADER ) );
+		colorFunctions.addAll( GraphToCsvUtils.getFeatureColorFunctions( projectModel ) );
+		return colorFunctions;
+	}
+
+	private static class ColorSchemeOptionsRenderer implements ListCellRenderer< ColorFunction >
+	{
+
+		private final DefaultListCellRenderer defaultRenderer = new DefaultListCellRenderer();
+
+		@Override
+		public Component getListCellRendererComponent( final JList< ? extends ColorFunction > list, final ColorFunction colorFunction,
+				final int index, final boolean isSelected, final boolean cellHasFocus )
+		{
+			if ( colorFunction.getGroup() == null )
```

Comment:
This condition (`colorFunction.getGroup() == null`) is true only if colorFunction is an EmptyColorFunction. So I would strongly suggest to make this intention explicit be writing:
```suggestion
			if ( colorFunction.getGroup() instanceof EmptyColorFuntion )
```